daemon/start.go

```diff
@@ -198,16 +199,32 @@ func (daemon *Daemon) containerStart(ctx context.Context, daemonCfg *configStore
 	if err != nil {
 		return setExitCodeFromError(container.SetExitCode, err)
 	}
+	defer func() {
+		if err != nil {
```

Comment:
Should this be `retErr` ?

```suggestion
		if retErr != nil {
```